### PR TITLE
[config/redesign] Redesign and refactor configuration flow

### PR DESCRIPTION
**Pull request by SalvatoreSantoro**
- [config/scripts] Redesign basy/python scirpts in python classes 
- [config/doc] Add doc for the new config structure
- [config/doc] Add UML documentation
- [config/configs] Rename configs by bus names
- [config/configs] Add structured configurability LOOPBACK from HBUS to MBUS
- [config] Rename clock domains by their source, e.g. MBUS_20, DDR4CH1_300
- [config] Rename RANGE_NAMES values with new naming convention
- [config] Add config_dump Makefile target
- [config] Remove redundant NUM_SI and NUM_MI from configs
- [hw/xilinx] Update paths IPs config tcl scripts
- [hw/xilinx] Update RTL for new net names
- [sw/SoC] Update C sources for new macro names

TODO:
- [ ] rebase on main
- [ ] Add doc for disabled address ranges (xbar acks)
- [ ] Implement #198
-----

# Test plan
## Nexys
TBD 

## Alveo au250
TBD

### CODE REVIEW

File: config/configs/hpc/config_hbus.csv [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Property,Value
+PROTOCOL,AXI4
+LOOPBACK,1
+ID_WIDTH,4
+NUM_SI,2
+NUM_MI,1
+MASTER_NAMES,MBUS s_acc
+RANGE_NAMES,DDR4CH_0
+RANGE_BASE_ADDR,0x80000
```
Comment (MaistoV) [inline config/configs/hpc/config_hbus.csv]: In this PR, we should try to max the DDR4CH_0 addressable space.

How does this impact the LOOPBACK constraints?

> NOTE: full DDR addressability issue remains for #193

File: config/doc/names.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# Naming Convention
+
+Each Node (peripheral or bus) exposes three distinct name categories.
```
Comment (MaistoV) [inline config/doc/names.md]: provide an example of naming here, as well

File: config/doc/names.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+- Identifies a specific address range belonging to a node.
+- Formed from the node’s FULL_NAME plus a range index:
+    `{FULL_NAME}_range_{id}`
+- If a node has a single address range, its RANGE_NAME equals its FULL_NAME.
+- Examples:
+    - `TIM_1_range_0`
+    - `TIM_1_range_1`
+- RANGE_NAMEs provide fine-grained reachability control; buses may reach some ranges but not others.
+
+The name is unfortunate because they COULD be different from the `RANGE_NAMES` expressed in a `.csv` file (for example if a node FULL_NAME is configured with multiple address ranges), they are
```
Comment (MaistoV) [inline config/doc/names.md]: Let's avoid to be self-denigratory, i.e. saying it is unfortunate.

It sufficies to say that RANGE_NAME is not user-specified, but it depends on the number of ranges requested. If there is only range_0, the suffix is omitted.

File: config/simply_v_conf/buses/bus.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+		#The number of ranges associated to each children node
+		self.CHILDREN_NUM_RANGES: int = data_dict["ADDR_RANGES"]
+        #Children Parameters internally used only to generate the children objects
+		#should never use them directly, after creating the new nodes, just refer to them to get
+		#ranges and clocks values
+		self._RANGE_NAMES: list[str] = data_dict["RANGE_NAMES"].copy()
+		self._RANGE_BASE_ADDR: list[int] = data_dict["RANGE_BASE_ADDR"].copy()
+		self._RANGE_ADDR_WIDTH: list[int] = data_dict["RANGE_ADDR_WIDTH"].copy()
+		#As default each bus just propagates its clock domain to its children
+		#MBUS is special since redefine this according to the values specified in its .csv
```
Comment (MaistoV) [inline config/simply_v_conf/buses/bus.py]: typo: redefines

File: config/simply_v_conf/buses/mbus.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+from general.addr_range import Addr_Ranges
+from .bus import Bus
+from general.env import Env
+from .nonleafbus import NonLeafBus
+from general.singleton import SingletonABCMeta
+
+#Only one MBUS should be created
+class MBus(NonLeafBus, metaclass=SingletonABCMeta):
+	env_global = Env.get_instance()
```
Comment (MaistoV) [inline config/simply_v_conf/buses/mbus.py]: Add a one-line comment introducing the use of Env

File: config/simply_v_conf/buses/nonleafbus.py [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+			if(not ((base_addr & (base_addr -1) == 0) and base_addr != 0)):
+				raise ValueError(f"Activated LOOPBACK in {self.FULL_NAME} with a BASE_ADDR"
+											" that isn't a power of 2.")
+	
+
+	
+	def _father_enable_loopback(self, child_name: str):
+		#NonLeaf buses that have "LOOPBACK" activated will call this function
+		#to modify the "father" bus informations to support the LOOPBACK
+		self.MASTER_NAMES.append(child_name)
```
Comment (MaistoV) [inline config/simply_v_conf/buses/nonleafbus.py]: clarify in doc that LOOPBACK implicitly adds a new SI that does not need to be in the original config CSV

File: config/simply_v_conf/buses/nonleafbus.py [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+
+	
+	def _check_legal_buses(self):
+		for b in self._children_buses:
+			if b.BASE_NAME not in self.LEGAL_BUSES:
+				raise ValueError(f"Unsupported bus {b.FULL_NAME} for this bus")
+
+
+	
+	# Function used to return children address ranges (of peripherals/buses) keeping an invariant
```
Comment (MaistoV) [inline config/simply_v_conf/buses/nonleafbus.py]: In general, try to avoid comments starting with "Fuctiuon used to" as this is redundant.
Comments should be direct and sufficiently summarize the functionality, e.g. "Return children address ranged .."

File: config/simply_v_conf/buses/nonleafbus.py [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+
+	def generate_children(self):
+		#NonLeaf buses need to create children peripherals and buses
+		peripherals_names: list[str] = []
+		peripherals_bases: list[int] = []
+		peripherals_widths: list[int] = []
+		peripherals_clock_domains: list[str] = []
+
+		for i, node_name in enumerate(self._RANGE_NAMES):
+			bases: list[int] = self._RANGE_BASE_ADDR[i:(i+self.CHILDREN_NUM_RANGES)]
```
Comment (MaistoV) [inline config/simply_v_conf/buses/nonleafbus.py]: Add a line of comment clarifying the use of CHILDREN_NUM_RANGES

File: config/simply_v_conf/buses/pbus.py [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+class PBus(LeafBus):
+	LEGAL_PERIPHERALS = Bus.LEGAL_PERIPHERALS + ("UART", "GPIOOUT", "GPIOIN", "TIM")
+	LEGAL_PROTOCOLS = Bus.LEGAL_PROTOCOLS + ("AXI4LITE",)
+
+	def __init__(self, base_name: str, data_dict: dict, asgn_addr_ranges: Addr_Ranges, clock_domain: str, 
+					clock_frequency: int):
+
+
+		axi_addr_width = 32
```
Comment (MaistoV) [inline config/simply_v_conf/buses/pbus.py]: Add a comment clarifying that these params are fixed for the PBUS

File: config/simply_v_conf/docs/busses.png
Comment (MaistoV) [inline config/simply_v_conf/docs/busses.png]: why are we keeping this class diagrams if we have the new ones? if these are generated, they should be gitignored

File: config/simply_v_conf/general/env.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+				return [0,1,2]
+			case "au50":
+				return [0,1]
+			case "Nexys-A7-100T-Master" | "Nexys-A7-50T-Master":
+				return [0]
+			case _:
+				raise ValueError("Unsupported Board configuration")
+
+
+	def get_supp_hbm(self) -> bool | None:
```
Comment (MaistoV) [inline config/simply_v_conf/general/env.py]: this addresses #192

File: config/simply_v_conf/general/simply_v.py [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+				return True
+
+		# return false if the bus wasn't found
+		return False
+
+
+
+	# Update the "hw" makefile with the corresponding "sys_targets" and "bus_targets" selected
+	# used from Vivado 
+	def config_xilinx_makefile(self, xilinx_makefile: str) -> None:
```
Comment (MaistoV) [inline config/simply_v_conf/general/simply_v.py]: for this method, add authors to file header:
```
# Author: Vincenzo Maisto <vincenzo.maisto2@unina.it>
# Author: Manuel Maddaluno <manuel.maddaluno@unina.it>
# Author: Stefano Mercogliano <stefano.mercogliano@unina.it>
# Author: Valerio Di Domenico <valerio.didomenico@unina.it>
```

File: config/simply_v_conf/peripherals/bram.py [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+from general.addr_range import Addr_Ranges
+from .peripheral import Peripheral
+
+class Bram(Peripheral):
+	def __init__(self, base_name: str, addr_ranges_list: Addr_Ranges, clock_domain: str, clock_frequency: int):
+
+		super().__init__(base_name, addr_ranges_list, clock_domain, clock_frequency)
+		self.IS_A_MEMORY = True
+
+	def config_ip(self, root_path: str, **kwargs) -> None:
```
Comment (MaistoV) [inline config/simply_v_conf/peripherals/bram.py]: for this, add Manuel Maddaluno and Vincenzo Maisto to authors of the file

File: config/simply_v_conf/peripherals/uart.py [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+import re
+
+class Uart(Peripheral):
+	def __init__(self, base_name: str, addr_ranges_list: Addr_Ranges, clock_domain: str, clock_frequency: int):
+
+		super().__init__(base_name, addr_ranges_list, clock_domain, clock_frequency)
+		self.HAL_DRIVER = True
+
+
+	def config_ip(self, root_path: str, **kwargs) -> None:
```
Comment (MaistoV) [inline config/simply_v_conf/peripherals/uart.py]: for this, add Manuel Maddaluno to authors

File: config/simply_v_conf/docs.sh [hunk 14]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
```
Comment (MaistoV) [inline config/simply_v_conf/docs.sh]: missing author and description

File: config/simply_v_conf/docs.sh
Comment (MaistoV) [inline config/simply_v_conf/docs.sh]: add doc for the use of this script in doc

File: config/simply_v_conf/main.py [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+
+from parsers.sys_parser import Sys_Parser
+from general.env import Env
+from general.logger import Logger
+from general.simply_v import SimplyV
+import argparse
+import traceback
+
+def parse_args():
+	parser = argparse.ArgumentParser()
```
Comment (MaistoV) [inline config/simply_v_conf/main.py]: name this object more explicitly, e.g. `arg_parser`, as it might get confused with the other parses

File: config/simply_v_conf/main.py [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+
+	parser.add_argument("--mode", required=True,
+	choices=[
```
Comment (MaistoV) [inline config/simply_v_conf/main.py]: tab/space this right for clarity

File: config/README.md [hunk 17]
```diff
@@ -63,7 +65,6 @@ The `VIO_RESETN_DEFAULT` parameter controls the programming-time value of core r
 | Name  | Description | Values | Default
```
Comment (MaistoV) [inline config/README.md]: Remove default column for #194

File: hw/units/custom_hls_conv_hbus/assets/conv_hbus/hw/conv_hbus/hls/impl/ip/drivers/krnl_conv_hbus_v1_0/src/xkrnl_conv_hbus_hw.h
Comment (MaistoV) [inline hw/units/custom_hls_conv_hbus/assets/conv_hbus/hw/conv_hbus/hls/impl/ip/drivers/krnl_conv_hbus_v1_0/src/xkrnl_conv_hbus_hw.h]: restore this

File: hw/xilinx/ips/common/xlnx_main_crossbar/config.tcl
Comment (MaistoV) [inline hw/xilinx/ips/common/xlnx_main_crossbar/config.tcl]: remove this

File: hw/xilinx/ips/common/xlnx_peripheral_crossbar/config.tcl
Comment (MaistoV) [inline hw/xilinx/ips/common/xlnx_peripheral_crossbar/config.tcl]: remove this

File: hw/xilinx/ips/embedded/xlnx_axi_uartlite/config.tcl [hunk 18]
```diff
@@ -5,7 +5,7 @@ create_ip -name axi_uartlite -vendor xilinx.com -library ip -version 2.0 -module
 set_property -dict [list                        \
                     CONFIG.C_BAUDRATE {9600}    \
                     CONFIG.C_DATA_BITS {8} \
-                    CONFIG.C_S_AXI_ACLK_FREQ_HZ {10000000} \
```
Comment (MaistoV) [inline hw/xilinx/ips/embedded/xlnx_axi_uartlite/config.tcl]: restore this

File: hw/xilinx/ips/hpc/xlnx_hbus_crossbar/.gitignore
Comment (MaistoV) [inline hw/xilinx/ips/hpc/xlnx_hbus_crossbar/.gitignore]: restore this

File: sw/SoC/examples/blinky/Makefile
Comment (MaistoV) [inline sw/SoC/examples/blinky/Makefile]: restore this, blinky must not rely on HAL or tinyIO

File: sw/SoC/lib/uninasoc/inc/plic.h [hunk 19]
```diff
@@ -10,7 +10,9 @@
 
 #include <stddef.h>
 #include <stdint.h>
-#include "uninasoc_conf.h"
+#include "uninasoc.h"
+
+// Import linker script symbol
```
Comment (MaistoV) [inline sw/SoC/lib/uninasoc/inc/plic.h]: remove tis comment

File: config/simply_v_conf/parsers/leafbus_parser.py [hunk 20]
```diff
@@ -0,0 +1,11 @@
+# Author: Salvatore Santoro <sal.santoro@studenti.unina.it>
+# Description: The class "LeafBus_Parser" inherits from the "Bus_Parser" class, extending the checked
+# properties with the ones common to all LeafBuses
+
+from .bus_parser import Bus_Parser
+
+class LeafBus_Parser(Bus_Parser):
+	#extend the father defined data structs used for parsing/validation
+
+	def __init__(self):
```
[the remark this answers is not in the document]
Comment (SalvatoreSantoro) [reply config/simply_v_conf/parsers/leafbus_parser.py]: I think i was troubleshooting something and just forgot them, the only one needing it is "LeafBus_Parser" since for now is and empty class, i'm removing the others.

File: config/configs/hpc/config_hbus.csv (hunk 1 above)
Comment (SalvatoreSantoro) [reply config/configs/hpc/config_hbus.csv]: I fixed the loopback implementation, addressing the edge cases in which the HBUS is the first or the last range of the MBUS and tries to activate LOOPBACK and tested this configuration (just checking the generated files since i can't really test it on hardware).
So the configuration was HBUS as first range on the MBUS, starting from address 0x0, DDR4CH_0 starting from 0x0 and LOOPBACK activated.
With an address width of 16 assigned to the HBUS, the AFTER range in the LOOPBACK will be 16 bit width (due to how the loopback feature is implemented to address Xilinx Interconnects constraints). 
So basically in this configuration the HBUS can address in the loopback configuration an AFTER address space as big as the space assigned to the DDR4CH_0.

File: config/configs/embedded/config_hbus.csv [hunk 21]
```diff
@@ -0,0 +1,10 @@
+Property,Value
+PROTOCOL,DISABLE
```
[the remark this answers is not in the document]
Comment (MaistoV) [reply config/configs/embedded/config_hbus.csv]: Great, we can discard the DISABLE feature, but this needs clear documentation and info-logging in the flow.
In particular, for each child bus of MBUS (or NonLeafBus in general) we should have a clear print clarifying which bus we are creating and which CSV we are parsing

File: config/configs/hpc/config_hbus.csv (hunk 1 above)
Comment (MaistoV) [reply config/configs/hpc/config_hbus.csv]: Ok, but for the current case (i.e. HBUS is not the first rage of MBUS) how large can we make the DDR4CH_0 address space and still loopback to the rest of the MBUS?
> NOTE: In the near future, we would most likely move to a default config with the HBUS at 0x8000000.

File: config/simply_v_conf/buses/nonleafbus.py [hunk 22]
```diff
@@ -0,0 +1,291 @@
...
+		self.MASTER_NAMES.append(child_name)
+		self.NUM_SI += 1
+
+	def _child_enable_loopback(self):
+		if(not self.father):
+			raise ValueError(f"Can't enable loopback on {self.FULL_NAME} without a father")
+
+		#NonLeaf buses that have "LOOPBACK" activated will call this function
+		#to modify themselves to support the LOOPBACK
+		#the function assumes ADDR_RANGES == 1, this invariant is kept by the
```
[the remark this answers is not in the document]
Comment (SalvatoreSantoro) [reply config/simply_v_conf/buses/nonleafbus.py]: Is actually enforced here https://github.com/SalvatoreSantoro/UninaSoC/blob/6ca67908ec65ed30320948c08de14119b02ed249/config/simply_v_conf/parsers/nonleafbus_parser.py#L18

File: config/simply_v_conf/factories/factory.py [hunk 23]
```diff
@@ -0,0 +1,69 @@
...
+# rather than an "idiomatic" factory method design pattern, in which there is
+# a concrete factory implementation for each concrete object to create, since
+# the creation logic right now is still simple.
+# Keep in mind that all the factories as defined are singletons, so they share
+# the same state, this is used to enforce "global" checks like
+# ensuring that there aren't some peripherals/buses with the same "FULL_NAME"
+# somewhere in the configuration because all the factories keep track of each object
+# created in the "ALREADY_CREATED" variable and are able to spot duplicated names
+# in the configuration
+# names MUST adhere the following syntax "{BASE_NAME}_{id}" where "{id} is a number.
```
[the remark this answers is not in the document]
Comment (SalvatoreSantoro) [reply config/simply_v_conf/factories/factory.py]: Done

File: config/simply_v_conf/main.py [hunk 24]
```diff
@@ -0,0 +1,125 @@
...
+	])
+
+	parser.add_argument("--system", required=True)
+	parser.add_argument("--mbus", required=True)
+	parser.add_argument("--pbus", required=True)
+	parser.add_argument("--hbus", required=True)
+	# Used only when configuring buses
+	parser.add_argument("--target_bus", required=False)
+
+	# everything after known args is treated as OUTPUT_FILES in the Makefile invocation
```
[the remark this answers is not in the document]
Comment (SalvatoreSantoro) [reply config/simply_v_conf/main.py]: Done

File: config/README.md [hunk 25]
```diff
@@ -101,46 +84,77 @@ After applying configuration changes to the target CSV files (`embedded` or `hpc
...
+The `MAIN_CLOCK_DOMAIN` is the clock domain of the core and the main bus (`MBUS`). All the slaves attached to the `MBUS` can have their own clock domain. If a slave has a domain different from the `MAIN_CLOCK_DOMAIN`, it needs a `xlnx_axi_clock_converter` to cross the clock domains. In this case the configuration flow will set the `<SLAVE_NAME>_HAS_CLOCK_DOMAIN` (i.e. `PBUS_HAS_CLOCK_DOMAIN`) variable which informs that the slave has its own clock domain.
 
 ### Scripting Architecture
-The directory `scripts/` holds multiple scripts, acting in the following scripting architecture:
+The directory `simply_v_conf/` holds all the configuration related files, the main configuration flow is depicted here:
 
-![Configuration flow](./doc/axi_xbar_config_with_check.png)
+![Configuration flow](./doc/simply_v.png)
 
 The multiple scripts generate outputs from common inputs:
```
[the remark this answers is not in the document]
Comment (SalvatoreSantoro) [reply config/README.md]: Done

File: config/doc/loopback.md [hunk 26]
```diff
@@ -0,0 +1,162 @@
...
+
+In the first case the child bus will use the new added slave interface to address the address space on its father `AFTER` its own address space.
+
+In the second case the child bus will use the new added slave interface to address the address space on its father `BEFORE` its own address space.
+
+In the third case the child bus will use the new added slave interface to address both the address space on its father `BEFORE` and `AFTER` its own. In order to accomodate two different address spaces through the same master port the child bus will also set the `ADDR_RANGES` parameter equal to **2** (this enables the addressing of two different address ranges for each **MASTER (M)** port on the bus).
+
+All the possible configurations are depicted in the following picture using the **MBUS** as a father bus and the **HBUS** as the child bus activating loopback.
+
+![Loopback](loopback.png)
```
Comment (MaistoV) [inline config/doc/loopback.md]: The order of examples in the picture is different from the doc. This can be confusing.
I suggest to be more versatile by having three different pictures, one for each case.

File: config/doc/loopback.md [hunk 27]
```diff
@@ -0,0 +1,162 @@
...
+
+In the case of setting `ADDR_RANGES = 2` the bus will also split all the address ranges of its children peripherals and buses (transparently to the user) in order to have two different ranges for each master port for every slave node, not only the "loopbacking" one (this is done since the `ADDR_RANGES` isn't port specific but has effect on all the master ports).
```
[the remark this answers is not in the document]
Comment (SalvatoreSantoro) [reply config/doc/loopback.md]: I think when I implemented the functionality I did the split of every node range in advance of possible problems, because PG059 never explicitly states that the "ADDR_RANGES" value is enforced on every MI slot, it just says that is a "Global" parameter and that is  the "Number of Address Ranges per MI slot" so is a bit vague regarding this, I don't really understand if that means that the ports are expected to match it or that is just like an "upper bound" value and less ranges are possible on the MI ports. In case this splitting action reveals to be useless while testing the new configuration for the hpc profile the part of the code in which the split happens could be removed without any problem, it shouldn't break anything really because it was just written as a memo for this ambiguous behaviour of the IP.

[the remark this answers is not in the document]
Comment (MaistoV) [reply config/doc/loopback.md]: You can play with any IP parameter in the Vivado GUI and trying to rebuild the IP to make sure it behaves as expected

File: hw/xilinx/ips/common/xlnx_peripheral_crossbar/config.tcl
Comment (SalvatoreSantoro) [reply hw/xilinx/ips/common/xlnx_peripheral_crossbar/config.tcl]: Done

File: sw/SoC/lib/uninasoc/inc/plic.h (hunk 19 above)
Comment (SalvatoreSantoro) [reply sw/SoC/lib/uninasoc/inc/plic.h]: Done

File: sw/SoC/examples/blinky/Makefile
Comment (SalvatoreSantoro) [reply sw/SoC/examples/blinky/Makefile]: Done

File: config/simply_v_conf/buses/nonleafbus.py [hunk 28]
```diff
@@ -0,0 +1,291 @@
...
+		#and activate the loopback for both child an father
+		if (self.LOOPBACK):
+			if(not self.father):
+				raise ValueError(f"Can't enable loopback on {self.FULL_NAME} without a father")
+
+			self.father._father_enable_loopback(self.FULL_NAME)
+			self._child_enable_loopback()
+
+		#Recursive call on all the buses
+		for bus in self._children_buses:
```
[the remark this answers is not in the document]
Comment (SalvatoreSantoro) [reply config/simply_v_conf/buses/nonleafbus.py]: Done

File: config/configs/embedded/config_hbus.csv (hunk 21 above)
[the remark this answers is not in the document]
Comment (SalvatoreSantoro) [reply config/configs/embedded/config_hbus.csv]: Changed the Makefile and main.py structure these are examples of the logging given by the application for embedded profile:

INFO: --- [CONFIG] Created BUS MBUS parsing file: /mnt/iso/home/salvatore/Progetti/UninaSoC/config/configs/embedded/config_mbus.csv ---
INFO: --- [CONFIG] Created BUS PBUS parsing file: /mnt/iso/home/salvatore/Progetti/UninaSoC/config/configs/embedded/config_pbus.csv ---
INFO: --- [CONFIG] Generated MBUS configs ---
INFO: --- [CONFIG] Generated PBUS configs ---
WARNING:  [CONFIG] HBUS configs NOT generated (bus disabled)
INFO: --- [CONFIG] Generated software configs ---
INFO: --- [CONFIG] Generated Xilinx configs ---
INFO: --- [CONFIG] Generated reachability dump ---
make[1]: uscita dalla directory «/mnt/iso/home/salvatore/Progetti/UninaSoC/config»

and hpc profile

INFO: --- [CONFIG] Created BUS MBUS parsing file: /mnt/iso/home/salvatore/Progetti/UninaSoC/config/configs/hpc/config_mbus.csv ---
INFO: --- [CONFIG] Created BUS PBUS parsing file: /mnt/iso/home/salvatore/Progetti/UninaSoC/config/configs/hpc/config_pbus.csv ---
INFO: --- [CONFIG] Created BUS HBUS parsing file: /mnt/iso/home/salvatore/Progetti/UninaSoC/config/configs/hpc/config_hbus.csv ---
INFO: --- [CONFIG] Created LOOPBACK range [0x0 - 0x7ffff] on MBUS from HBUS ---
INFO: --- [CONFIG] Created LOOPBACK range [0x90000 - 0x9ffff] on MBUS from HBUS ---
INFO: --- [CONFIG] Generated MBUS configs ---
INFO: --- [CONFIG] Generated PBUS configs ---
INFO: --- [CONFIG] Generated HBUS configs ---
INFO: --- [CONFIG] Generated software configs ---
INFO: --- [CONFIG] Generated Xilinx configs ---
INFO: --- [CONFIG] Generated reachability dump ---

[the remark this answers is not in the document]
Comment (SalvatoreSantoro) [reply config/configs/embedded/config_hbus.csv]: Done
